torch/_inductor/ir.py

```diff
@@ -8508,6 +8508,35 @@ def __init__(
         self.name = V.graph.register_buffer(self)
         V.graph.register_operation(self)
 
+    @staticmethod
+    def _clone_aliased_inputs(carried_inputs: Sequence[IRNode]) -> Sequence[IRNode]:
```

Comment:
Can you add a test that directly invokes this logic?

Reply:
Added a scan test. Can only trigger the cse in partitioner. So the following inference cannot trigger cse:
```python
import torch
def f(x):
    a = torch.zeros_like(x)
    b = torch.zeros_like(x)
    def cond_fn(it, a, b):
        return it < 2

    def body_fn(it, a, b):
        return it + 1, a + 1, b + 1

    it, new_a, new_b = torch._higher_order_ops.while_loop(cond_fn, body_fn, (torch.tensor(0, dtype=torch.int64), a, b))
    return new_a + new_b + it

out = torch.compile(f)(torch.randn(3, 4, requires_grad=True))
```